control-plane/pkg/reconciler/source/source.go

```diff
@@ -170,6 +178,19 @@ func (r Reconciler) reconcileConsumerGroup(ctx context.Context, ks *sources.Kafk
 		Spec:       expectedCg.Spec,
 		Status:     cg.Status,
 	}
+	newCg.Annotations = expectedCg.Annotations
+
+	// If KEDA is not enabled, then we must update the ConsumerGroup replicas to match KafkaSource consumers
+	// TODO: code below failing unit tests with err: "panic: interface conversion: testing.ActionImpl is not testing.GetAction: missing method GetName"
+	/*if err := discovery.ServerSupportsVersion(r.KubeClient.Discovery(), keda.KedaSchemeGroupVersion); err != nil {
+		newCg.Spec.Replicas = ks.Spec.Consumers
+	}*/
+
+	_, err = r.KedaClient.KedaV1alpha1().ScaledObjects(cg.GetNamespace()).List(ctx, metav1.ListOptions{})
+	if err != nil && apierrors.IsNotFound(err) {
```

Comment:
something like:
```suggestion
	if r.isKEDAEnabled() {
```
would be an excellent approach to encapsulate this logic (it can be done in a follow-up PR).

Another benefit of this encapsulation is that we can also introduce a flag to enable/disable KEDA in addition to the detection to have more control.

Reply:
+1

Reply:
Absolutely will help. Coming in a forthcoming PR.